### DISCUSSION TITLE
DataTable - sortable header with InputText for SQL site filtering

### DISCUSSION THREAD

**Discussion by duese1990** [primefaces/community#684] [PrimeVue] [answered]
Hi,

I'm using the DataTable with an InputText-Field inside the #header-Element for filtering on SQL database site. Additionally the sortable paramater is active and the sort-icon is visible.

The issue is, that each time I focus the InputText the sort function will be triggered. I just like to have the sort function on the sort-icon and not on the complete header.

Is there an solution?

Example column:
```
<Column field="machineNo" sortable >
<template #header>
<span class="p-float-label" >
<InputText id="machineNo" v-model="searchMachineNo" v-on:change="getReferences(true)"/>
<label for="machineNo">Maschine</label>
</span>
</template>
<template #body="slotProps">
{{ slotProps.data.machineNo }}
</template>
</Column>
```

Kind regards,
duese1990

**Comment by tugcekucukoglu** [comment, marked as answer]
Hi,

It seems there is a bug here. It will be fixed for the next version. https://github.com/primefaces/primevue/issues/4988

Thank you.
